core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/TypeUtils.kt

```diff
@@ -164,20 +175,30 @@ internal fun commonParents(classes: Iterable<KClass<*>>): List<KClass<*>> =
                     it.size == 1 && it[0].visibility == KVisibility.PUBLIC -> { // if there is only one class - return it
                         listOf(it[0])
                     }
+
                     else -> it.fold(null as (Set<KClass<*>>?)) { set, clazz ->
                         // collect a set of all common superclasses from original classes
                         val superclasses =
-                            (clazz.allSuperclasses + clazz).filter { it.visibility == KVisibility.PUBLIC }
-                        set?.intersect(superclasses) ?: superclasses.toSet()
+                            (clazz.allSuperclasses + clazz).filter { it.visibility == KVisibility.PUBLIC }.toSet()
+                        set?.intersect(superclasses) ?: superclasses
                     }!!.let {
-                        it - it.flatMap { it.superclasses } // leave only 'leaf' classes, that are not super to some other class in a set
+                        it - it.flatMap { it.superclasses }
+                            .toSet() // leave only 'leaf' classes, that are not super to some other class in a set
                     }.toList()
                 }
             }
         }
-    }
+    }.sortedBy { it.simpleName } // make sure the order is stable to avoid bugs
 
-internal fun baseType(types: Set<KType>): KType {
+/**
+ * Returns the common type of the given [types].
+ *
+ * @param types the types to find the common type for
+ * @param listifyValues if true, then values and nulls will be wrapped in a list if they appear among other lists.
+ *   For example: `[Int, Any?, List<Int>]` will become `List<Int>` instead of `Any?`. If there is another collection
```

Comment:
is that so? 
It seems like a bug. Because type of null should be Nothing?

Reply:
Definitely not `Nothing?` right? That only appears for thrown errors and return statements. `Nothing?` cannot have an instance. `Int, Any?, List<Int>` will become `Any?` as far as I know.

Reply:
Nothing appears for thrown errors, but `Nothing?` is a special type for null values 
![image](https://user-images.githubusercontent.com/12936457/202169328-c55f7497-f7c4-4a3e-8916-ee6f89c12c31.png)

Reply:
Oh, I see now. I made an error in the KDoc. `[Int, Any?, List<Int>]` becomes `List<Any?>?`.
`[Int, Nothing?, List<Int>]` will become `List<Int>` (or at least, it should, but it doesn't atm... Bug :)  )

Reply:
All handled in latest commit :)